src/products/service/__test__/product.service.spec.ts

```diff
@@ -0,0 +1,212 @@
+import { ConflictException, NotFoundException } from '@nestjs/common';
+import { DataSource, QueryRunner } from 'typeorm';
+import { Test, TestingModule } from '@nestjs/testing';
+
+import { Product } from '../../domain/product.entity';
+import { ProductCategory } from '../../domain/product-category.entity';
+import { ProductCategoryService } from '../product-category.service';
+import { ProductRepository } from '../../repository/product.repository';
+import { ProductService } from '../product.service';
+import { StockDataDto } from '../../dto/stock/stock-data.dto';
+import { StockService } from '../stock.service';
+import { Supplier } from 'src/products/enum/supplier.enum';
+
+jest.mock('../../repository/product.repository');
+jest.mock('../product-category.service');
```

Comment:
**suggestion:** you should try to have / create this mocks in a shared placed so you can use them in other possible unit tests.

You also have to option to create special mock classes which extend the implementation of the injectables and use them in `createTestingModule` method in the `providers` with `{provide: <X>, useClass: MyMockClass}` or `{provide: <X>, useFactory: MyFactoryMock}`